src/main/java/core/basesyntax/service/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitShopDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.handler.OperationHandler;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private static final String DEFAULT_START = "fruit,quantity" + System.lineSeparator();
+    private static final String COMMA = ",";
+    private static final int INDEX_OF_OPERATION_TYPE = 0;
+    private static final int INDEX_OF_FRUIT = 1;
+    private static final int INDEX_OF_QUANTITY = 2;
+    private FruitShopDao fruitShopDao;
+    private OperationStrategy operationStrategy;
+
+    public FruitShopServiceImpl(FruitShopDao fruitShopDao, OperationStrategy operationStrategy) {
+        this.fruitShopDao = fruitShopDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void addData(List<String> data) {
+        for (int i = 1; i < data.size(); i++) {
+            String[] dataFromList = data.get(i).split(COMMA);
```

Comment:
you have to check that size is of dataFromList is 3, because now you can't guarantee that you will not get indexOutOfBound if length will be 2 or 1

Reply:
but we receive valid data

Reply:
WHo told you that? this is example file in test file or in real life you have to check before using

Reply:
fixed